src/client/Microsoft.Identity.Client/WwwAuthenticateParameters.cs

```diff
@@ -60,6 +60,13 @@ public IEnumerable<string> Scopes
         /// </summary>
         public string Authority { get; set; }
 
+        /// <summary>
+        /// AAD tenant ID to which the Azure subscription belongs to.
+        /// </summary>
+        public string TenantId => Instance.Authority
```

Comment:
@bgavrilMS : I'm not a big fan of this solution to extract the tenant

WWWAuthenticateParameters was a small class that we could have split into a small assembly that SAL and MISE could use (discussed with @brentschmaltz).
Does `Authority` brings a more dependencies?

Reply:
Ok, I was not aware of this ask. 

We could split out the logic of parsing the authority URL and exposing the tenant ID. But I must warn that parsing works today because MSAL deals only with AAD, ADFS and B2C and can tell the authorities apart. If SAL and MISE want to support any IdP, extracting the `tenantid` will not be as simple. 

If we were to extract a package with WWWAuthenticateHeaders, I would move the TenantId to be an extension method in MSAL. Sadly C# does not support extension properties, so I propose that we change this in "GetTenantId" (a method is also more consistent with the fact that there is non-trivial work done there). Would this work @jmprieur  @abatishchev ?

Reply:
Let me make it a method, so we could easier convert it to an extension method in the future, without introducing breaking changes or anything.